crates/bevy_ecs/src/system/query.rs

```diff
@@ -239,11 +244,16 @@ use core::{
 /// # struct ComponentB;
 /// # fn system(
 /// // This will not panic.
+/// // This creates a perfect separation where:
+/// // 1. First query reads entities that have ComponentA
+/// // 2. Second query modifies ComponentB only on entities that DON'T have ComponentA
+/// // Result: No entity can ever be accessed by both queries simultaneously
 /// query_a: Query<EntityRef, With<ComponentA>>,
 /// query_b: Query<&mut ComponentB, Without<ComponentA>>,
 /// # ) {}
 /// # bevy_ecs::system::assert_system_does_not_conflict(system);
 /// ```
+/// The core principle is immutable: `EntityRef` can read everything on an entity, so it must be completely separated from any mutable access. The `With`/`Without` filters achieve this by ensuring the queries operate on entirely different sets of entities.
```

Comment:
"The core principle is immutable" doesn't make sense to me. Can you try rephrasing? Rest looks good.

Reply:
done